Fix test failures on MariaDB

junaruga opened the pull request:
This PR fixes #965 and fixes #1152 test failures related to MariaDB.

Please check each commit's comment.
Thanks.

sodabrew reviewed (approved):
Looks excellent, thank you for running this down!

sodabrew commented:
There are still some test failures after removing the last allow-fail, though?

junaruga commented:
> There are still some test failures after removing the last allow-fail, though?

For cases I removed the allow-fail, there is no test failures any more.

But there are still roughly 3 test failures in other CI cases.

1.

```
- {os: ubuntu-20.04, ruby: 2.4, db: mariadb10.3, allow-failure: true} # => it will be fixed after the PR #1183 will be merged.
```

2.

```
# Allow failure due to Mysql2::Error: Unknown system variable 'session_track_system_variables'.
- {os: ubuntu-16.04, ruby: 2.4, db: mariadb10.0, allow-failure: true} # => we might be fix this case with the same way on the PR #1183.
```

3.

```
# Allow failure due to the issue #1165.
- {os: ubuntu-20.04, ruby: 2.4, db: mysql80, allow-failure: true}
- {os: macos-latest, ruby: 2.4, db: mysql, allow-failure: true}
```

junaruga commented:
Travis mysql55 started error since yesterday due to the package repository not found.
https://travis-ci.org/github/brianmario/mysql2/jobs/767276558

```
E: Failed to fetch https://downloads.apache.org/cassandra/debian/dists/39x/main/binary-amd64/Packages  404  Not Found
E: Failed to fetch https://downloads.apache.org/cassandra/debian/dists/39x/main/binary-i386/Packages  404  Not Found
```

junaruga commented:
5. macOS MariaDB case `Mysql2::EM::Client` fails. I think it rarely happens. I saw it in the past.
https://github.com/brianmario/mysql2/pull/1193/checks?check_run_id=2363155750#step:9:536

junaruga commented:
> So, I can add the allow_failure to the macOS MariaDB case again if you like.

I rebased the last allow-fail's commit keeping allow_failure to the macOS MariaDB case.
I also added one more commit to add allow_failures to mysql55 case in Travis for now.

junaruga commented:
> I also added one more commit to add allow_failures to mysql55 case in Travis for now.

I think an possible alternative of mysql55 could be mysql56 on xenial or trusty if the mysql55's repo issue will be not fixed.

http://repo.mysql.com/apt/ubuntu/dists/xenial/
http://repo.mysql.com/apt/ubuntu/dists/trusty/

junaruga commented:
@sodabrew ping.

junaruga commented:
Finally travis-ci.org shut down will occur on May 31, 2021. You can see https://github.com/brianmario/mysql2/issues/1149#issuecomment-835549157 for detail.

sodabrew commented on spec/mysql2/client_spec.rb:
```diff
@@ -488,6 +494,7 @@ def run_gc
   end
 
   it "should set custom connect_attrs" do
+    skip("DON'T WORRY, THIS TEST PASSES - but PERFORMNCE SCHEMA is not enabled in your MySQL daemon.") unless performance_schema_enabled
```
```suggestion
    skip("DON'T WORRY, THIS TEST PASSES - but PERFORMANCE SCHEMA is not enabled in your MySQL daemon.") unless performance_schema_enabled
```

sodabrew commented on spec/mysql2/client_spec.rb:
```diff
@@ -479,6 +484,7 @@ def run_gc
   end
 
   it "should set default program_name in connect_attrs" do
+    skip("DON'T WORRY, THIS TEST PASSES - but PERFORMNCE SCHEMA is not enabled in your MySQL daemon.") unless performance_schema_enabled
```
```suggestion
    skip("DON'T WORRY, THIS TEST PASSES - but PERFORMANCE SCHEMA is not enabled in your MySQL daemon.") unless performance_schema_enabled
```

junaruga commented:
Thanks for finding my typos and the merging!
